### DISCUSSION TITLE
Adding a custom submodel for LAM

### DISCUSSION THREAD

**Discussion by DarioSlaifsteinSk** [pybamm-team/PyBaMM#3912] [Q&A]
Hi,
I have a reduced order model (ROM) for loss of active material (LAM) in the anode and I want to incorporate it to an SPM model.
The model is from [Jin (2022) ](https://www.sciencedirect.com/science/article/pii/S0013468621019356?via%3Dihub) with the following Eqs.: 

$$
Q_{AM,t} = Q_{Li, 0} . \int_0^t  k_{AM}.e^{\frac{-E_{AM}}{R.T}}.SoC_{t}.|i_{t}| .dt
$$

where:

$$
k_{\text{AM}}=\frac{k_{\text{AM}}^0}{\varepsilon_{\text{AM}, 0}}
$$

$$
SoC_t = \frac{x_{100} - x_t }{x_{100} - x_{0}}
$$

the parameters are only $k_{\text{AM}}$ in [1/Ah] and $E_{\text{AM}}$.
I checked this [notebook](https://docs.pybamm.org/en/latest/source/examples/notebooks/models/using-submodels.html) but i don't really get the logic of it.

I'm guessing that I should do something like the following:
```python
# initialize the models
model = pybamm.lithium_ion.SPM() # cell model
LAMmodel = pybamm.active_material.BaseModel()

# LAM model parameters
kAM = pybamm.Parameter("LAM Reaction rate constant [1 .A-1.h-1]")
EAM = pybamm.Parameter("LAM Activation energy [J.mol-1]")

# LAM model variables
QAM = pybamm.Variable("Capacity loss of active material [A.h]")
dQAMdt = kAM * pybamm.exp(-EAM / (pybamm.constants.R * T) ) * abs(Iapp) * Cell_SOC * Q0;

LAMmodel.rhs = {QAM: dQAMdt}
model.submodels["negative primary active material"] = LAMmodel
```
The cell current `Iapp`, the initial capacity `Q0`, the cell temperature `T`, the state of charge `Cell_SOC` and the lost capacity `QAM` should all come from the cell `model`.  But... I have no idea how to do it. 😅 

I'm "using" the "reaction rate" option for LAM in the SPM but I have no reference values for my cell so all my results are kinda dubious.

Thanks in advance,

**Comment by valentinsulzer** [comment]
If you haven't done this yet I would start by making a copy of `BasicSPM` and adding the new equations in directly https://github.com/pybamm-team/PyBaMM/blob/develop/pybamm/models/full_battery_models/lithium_ion/basic_spm.py

It won't work with experiments but we're working on making that possible in the next release

**Reply by DarioSlaifsteinSk** [reply]
Well I literally needed it for that so...
A little context, I´m using PBROMs (performance and ageing) for control and I wanted to investigate the effect of using old and fresh batteries, which is one of the main advantages of the models.

I wanted to have a good/realistic aged battery dataset for the initial conditions of the cell and controller.

I´ll use what I can find in the literature then and wait for the new features. If these are incorporated that´d be great.
